tapchannel/aux_leaf_creator.go

```diff
@@ -25,67 +23,12 @@ const (
 	DefaultTimeout = 30 * time.Second
 )
 
-// LeafCreatorConfig defines the configuration for the auxiliary leaf creator.
-type LeafCreatorConfig struct {
-	ChainParams *address.ChainParams
-}
-
-// AuxLeafCreator is a Taproot Asset auxiliary leaf creator that can be used to
-// create auxiliary leaves for Taproot Asset channels.
-type AuxLeafCreator struct {
-	startOnce sync.Once
-	stopOnce  sync.Once
-
-	cfg *LeafCreatorConfig
-
-	// ContextGuard provides a wait group and main quit channel that can be
-	// used to create guarded contexts.
-	*fn.ContextGuard
-}
-
-// NewAuxLeafCreator creates a new Taproot Asset auxiliary leaf creator based on
-// the passed config.
-func NewAuxLeafCreator(cfg *LeafCreatorConfig) *AuxLeafCreator {
```

Comment:
FWIW, I think we could've kept the struct structure for ease of use/isolation, in say tests.